challenges-7-plus.yml

```diff
@@ -0,0 +1,277 @@
+    so that we can have a cleaned preview, doesn't quite work. When we came
+    back to the page, we saw a greyed invalidation div covering it!
+    
+    Fixing this required some more hacking on our part in order to have it working
+    properly.
+    
+    Why did we get this weird behavior?
+  explanation: |
+    The `turbo:before-cache` event is synchronous, which means that what you do in that
+    event handler will happen instantly. If in that event you call some animation
+    (like the bootstrap `close` function), then the animation won't finish executing,
+    preventing any callbacks after the animation from cleaning up the page.
+    
+    It's important to know this for any third party code that we use, so that when
+    preparing a snapshop on the `before-cache` event goes smoothly!
+    
+    > Alternatively, you can disable cache on certain pages if the clean up
+    > process seems too complicated
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          This happened because bootstrap has a bug where some times calling `close`
+          on a modal doesn't close it properly.
+        b: |
+          This happened because Turbo has a bug where some times our code inside a
+          `turbo:before-cache` event doesn't execute properly.
+        c: |
+          This happenbed because we called `close` on our dialog and forgot to make the
+          `turbo:before-cache` callback asynchronous.
+        d: |
+          This happened because we called `close` on our dialog right inside
+          `turbo:before-cache` and the closing animation didn't have time to
+          finish before the cache was saved.
+      correct: d
+
+js_widget:
+  question: |
+    In this video, we've added a Weather Widget sanippet into our page and saw how it
+    only work the first time our page loaded. When we clicked on subsecuent pages, it
+    stopped working!
+    
+    Why is this so?
+  explanation: |
+    In our video, the snipped of code is executed every time we load a page because it's
+    in our `body` element! But in it there's a check before adding the `script` to our
+    `head`. This `script` is only added if it's not already present. This is why the
+    `script` tag was only added once, and therefore, only executed once between our page
+    views!
+    
+    > When we forced our snippet to add the `script` tag without any check, it effectively
+    > added a new copy of itself into our `head` *every* single time we change the page.
+    > This worked, but now we have a new problem: Our `head` is growing forever!
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          Because the `script` tag added by our snippet of code is only added once, hence
+          it only executes once.
+        b: |
+          Because the snippet of code added by us is only executed once, hence it only
+          adds the `script` tag once
```

Comment:
Hmm, I'm having a hard time distinguishing between A and B.

> `script` tag added by our snippet
vs
> snippet of code adde by us

Reply:
Rephrasing it to:

```
        a: |
          Because the `<script>` tag added by our snippet of code is only added once, hence
          it only executes once.
        b: |
          Because the snippet of code added by us to the `<bode>` element is only executed once,
          hence it only adds the `<script>` tag to our `<head>` once.
```